### PR TITLE
Add "strip line breaks" button to multiline paste warning dialog

### LINKED ISSUE #9400: Add a third option which replaces newlines by spaces when pasting content with multiple lines
# Description of the new feature/enhancement

When pasting in the Terminal a command line that ends with a line break, my options are either to paste and run the command, or abort the operation. Most of the times, I actually want to paste while trimming the trailing line break, such that I can edit the command I am pasting.

image

# Proposed technical implementation details (optional)

When pasting, a new button would appear in the confirmation dialog, that would allow to paste while trimming line breaks (if the string contains line breaks that are not trimmed, replace them by a space?)

### PR DESCRIPTION
**Pull request by Nacimota**
## Summary of the Pull Request
Adds a button to the multiline paste warning dialog that replaces line breaks (`\r\n`, `\r`, and `\n`) in the clipboard text with spaces before pasting.

## PR Checklist
* [x] Closes #9400
* [x] CLA signed.
* [ ] Tests added/passed
* [ ] Documentation updated. If checked, please file a pull request on [our docs repo](https://github.com/MicrosoftDocs/terminal) and link it here: #xxx
* [ ] Schema updated.
* [ ] I've discussed this with core contributors already. If not checked, I'm ready to accept this work might be rejected in favor of a different grand plan. Issue number where discussion took place: #xxx

## Detailed Description of the Pull Request / Additional comments
Sets the `SecondaryButtonText` property on the `MultiLinePasteDialog` via the `Resources.resw` file. Adds a condition in `TerminalPage::_PasteFromClipboardHandler` to check if the secondary button was pressed and to replace line breaks in the text with spaces, if so.

multiline-paste-dialog with three buttons

C++ newbie, feedback welcome.

### Questions for reviewers:
1. Is the new button text (_"Strip line breaks"_) acceptable? Should any of the other text in the dialog be changed?
3. Is the stripping code appropriate? Should we trim the end of the string as well (I'm not sure if there's a preferred trim function)?

## Validation Steps Performed
Manually tested all three buttons with multiline text pastes using `\r\n`, `\r` and `\n` in PowerShell, Command Prompt, and Bash (WSL).

### CODE REVIEW

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw [hunk 1]
```diff
@@ -447,6 +447,9 @@
   <data name="MultiLinePasteDialog.PrimaryButtonText" xml:space="preserve">
     <value>Paste anyway</value>
   </data>
+  <data name="MultiLinePasteDialog.SecondaryButtonText" xml:space="preserve">
+    <value>Strip line breaks</value>
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/Resources/en-US/Resources.resw:451]:
```suggestion
    <value>Paste without line breaks</value>
```
Maybe? I feel like we still want "paste" in there so that it's clear that you're pasting and not just modifying the shown text.

@cinnamon-msft Thoughts?

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw [hunk 2]
```diff
@@ -447,6 +447,9 @@
   <data name="MultiLinePasteDialog.PrimaryButtonText" xml:space="preserve">
     <value>Paste anyway</value>
   </data>
+  <data name="MultiLinePasteDialog.SecondaryButtonText" xml:space="preserve">
+    <value>Strip line breaks</value>
+  </data>
```
Comment (carlos-zamora) [inline src/cascadia/TerminalApp/Resources/en-US/Resources.resw:452]: Also, could you add a comment tag specifying that this is a button label that strips the line breaks and pastes the presented text? Recently got feedback from the localization team that we should add more comments in and I'm now always on edge hahaha 😅

**Review by carlos-zamora — CHANGES_REQUESTED** [review]
Really just want the localized string figured out. Everything else looks great! Thanks! 😊

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw (hunk 1 above)
Comment (Nacimota) [reply src/cascadia/TerminalApp/Resources/en-US/Resources.resw:451]: I originally had _"Strip line breaks and paste"_ but that felt way too wordy; I do think _"Paste without line breaks"_ is much nicer.

Comment (Nacimota) [reply src/cascadia/TerminalApp/Resources/en-US/Resources.resw]: Hmm, I seem to get cut-off with the longer text:

image

Thoughts?

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw (hunk 2 above)
Comment (Nacimota) [reply src/cascadia/TerminalApp/Resources/en-US/Resources.resw:452]: Sure thing. I'll add one for the primary button as well if that helps; let me know if the wording is unclear.

File: src/cascadia/TerminalApp/Resources/en-US/Resources.resw (hunk 1 above)
Comment (HermasciousLevander) [reply src/cascadia/TerminalApp/Resources/en-US/Resources.resw]: Perhaps simply "_Remove line breaks_" would work? I can't really think of a good way to say paste without line breaks without making it too long. "_Paste w/o line breaks_" might work, but that might look weird.

**Comment by zadjii-msft** [comment]
Hey sorry for the runaround on this one. Talked with the team today - we do feel like the right way forward is just to trim the trailing newline if that's the only one in the paste contents. To quote: "It doesn't make sense to be warned for multiple lines, but copy a single line and try to paste it AND ONLY SEE ONE LINE IN THE DIALOG. That's dumb".

So thanks for the gusto in implementing the original request, but I think we want to take this another direction!
